src/OpenSage.Game/Logic/Object/Update/ProductionUpdate.cs

```diff
@@ -232,6 +232,18 @@ private void GrantUpgrade(UpgradeTemplate upgradeDefinition)
 
         internal bool CanProduceObject(ObjectDefinition objectDefinition)
         {
+            // only one hero of the same kind can be produced at a time
+            if (objectDefinition.KindOf.Get(ObjectKinds.Hero))
+            {
+                foreach (var job in ProductionQueue)
+                {
+                    if (job.ObjectDefinition.Name == objectDefinition.Name)
```

Comment:
Same here - do we not have referential equality for `ObjectDefinition` instances?

Reply:
fixed